programs/client/TestHint.cpp

```diff
@@ -0,0 +1,105 @@
+#include "TestHint.h"
+
+#include <Common/Exception.h>
+#include <Common/ErrorCodes.h>
+#include <IO/ReadBufferFromString.h>
+#include <IO/ReadHelpers.h>
+#include <Parsers/Lexer.h>
+
+namespace
+{
+
+/// Parse error as number or as a string (name of the error code const)
+int parseErrorCode(DB::ReadBufferFromString & in)
+{
+    int code = -1;
+    String code_name;
+
+    auto * pos = in.position();
+    tryReadText(code, in);
```

Comment:
Numbers with letters in the parsed as codes. Should we fix it?
```
clickhouse-client --testmode -n -q 'select 1 -- { serverError 100i }'
1
The query succeeded but the server error '100' was expected.
```

Reply:
I left this because this will allow trailing semicolon out of the box, i.e. `select 1 -- { serverError 100; }`, since there are such bits in tests.
Sure I can fix this too allow only trailing semicolon, but this will make the logic a little bit complex and I'm not sure that this worth it, since this logic is used only for tests (i.e. it is not major bits).

Reply:
Ok, let's leave as is